### ISSUE TITLE
[ROCm] Sparse-MLA decode top-k falls back to the vLLM kernel above 64k because of an aiter 0.1.19 limitation

### ISSUE THREAD

**Issue by stefanskiasan** [vllm-project/vllm#55615]
**[ROCm] Sparse-MLA decode top-k falls back to the vLLM kernel above 64k
because of an aiter 0.1.19 limitation**

`vllm/v1/attention/ops/rocm_aiter_mla_sparse.py` disables the AITER top-k
kernel for decode:

```python
if not is_prefill:
    # AITER v0.1.19 decode is one-block only. This measured gfx950
    # FP32/k=1024 compressed-row boundary is independent of the native
    # split-count boundary in sampler.cu.
    if (
        num_rows <= _GFX950_C4A_NATIVE_MAX_ROWS            # 256
        and max_valid_seq_len > _GFX950_C4A_AITER_MAX_COMPRESSED_SEQ_LEN  # 64 KiB
    ):
        return None
```

The comment states the limitation is specific to **aiter v0.1.19**. On
**0.1.21.post1** the AITER kernel handles these shapes and is faster.

**Measured** (MI355X / gfx950, ROCm 7.2.3, aiter 0.1.21.post1, 40 rows —
that is 8 concurrent requests × MTP-5 — `topk=2048`, FP32 logits; three
repeats, median, 8 warm-up calls, synchronised timing windows):

| context | `torch.ops._C` (current path) | `aiter.ops.topk` | delta |
|---:|---:|---:|---:|
| 65 536 | 0.0505 ms | 0.0383 ms | **−24.1 %** |
| 131 072 | 0.0785 ms | 0.0624 ms | **−20.5 %** |

Both kernels select the **identical set** of indices — 100 % overlap on the
checked row, so this is not a precision trade-off.

Note the condition also has a somewhat counter-intuitive shape: it disables
AITER precisely for **small** batches (`num_rows <= 256`) at long context,
which is the low-concurrency long-context case — exactly where per-token
latency matters most to a user.

Suggestion: gate on the installed aiter version rather than unconditionally,
or drop the branch once 0.1.21 is the floor. Happy to run any variant on
gfx950.

For transparency about the size of the effect: in our deployment this kernel
accounts for ~1.1 % of decode TPOT, so the end-to-end gain is small. The
reason to fix it is that the stated justification no longer holds.
